### ISSUE TITLE
how to get fine_tune model output probability

### ISSUE THREAD

**Issue by policeme** [google-research/bert#322]
hi, I have a problem, when i finish bert fine tune model of run_classifier.py, I use it to my task, but i don't know how to get probability from fine tune model.
when I use below code, I get not same result when I feed same text_a and text_b, Please help me! Thanks!

   
![Uploading image.png…]()

**Comment by policeme** [comment]
def predict(self, text_a, text_b):
        """
        bert model predict
        :param text_a: text a
        :param text_b: text b
        :return: label, similarity
        """
        input_ids, input_mask, segment_ids = self.convert_single_example(text_a, text_b)
        input_ids = np.reshape(input_ids, [-1, len(input_ids)])
        input_mask = np.reshape(input_mask, [-1, len(input_mask)])
        segment_ids = np.reshape(segment_ids, [-1, len(segment_ids)])

        output_layer = self.sess.run(self.model.get_pooled_output(), feed_dict={
            self.input_ids: input_ids,
            self.input_mask: input_mask,
            self.input_type_ids: segment_ids
        })

        logits = tf.matmul(output_layer, self.output_weights, transpose_b=True)
        logits = tf.nn.bias_add(logits, self.output_bias)
        probabilities = tf.nn.softmax(logits, axis=-1)

        print(self.sess.run(probabilities))

**Comment by policeme** [comment]
could someone can help me, Thanks you @jacobdevlin-google

**Comment by haleemason** [comment]
Maybe try something like this:

```python
def getPrediction(in_sentences):
    labels = ["Negative", "Positive"]
    input_examples = [run_classifier.InputExample(guid="", text_a = x, text_b = None, label = 0) 
                      for x in in_sentences] # here, "" is just a dummy label
    input_features = run_classifier.convert_examples_to_features(input_examples, label_list, MAX_SEQ_LENGTH, tokenizer)
    predict_input_fn = run_classifier.input_fn_builder(features=input_features, 
                                                       seq_length=MAX_SEQ_LENGTH,
                                                       is_training=False, drop_remainder=False)
    predictions = estimator.predict(predict_input_fn)
    return [(sentence, prediction['probabilities'], labels[prediction['labels']]) for sentence, prediction in zip(in_sentences, predictions)]
```

Then you can pass it a list of examples:
```python
pred_sentences = [
  "That movie was absolutely awful",
  "The acting was a bit lacking",
  "The film was creative and surprising",
  "Absolutely fantastic!"
]
predictions = getPrediction(pred_sentences)
```

This example is from the example notebook: https://colab.research.google.com/github/google-research/bert/blob/master/predicting_movie_reviews_with_bert_on_tf_hub.ipynb

**Comment by jinamshah** [comment]
Hi, this gives me negative probabilities like in the example. Can someone tell me how to get positive probabilities from this. So that thresholding can be applied to this.
A use case for this would be for bi-class classification and having an input that doesn't belong to any class
